feat(route): 增加大数据专家委员会和中国人工智能学会新闻

tudou027 opened the pull request:
<!-- 
如有疑问，请参考 https://github.com/DIYgod/RSSHub/discussions/8002
Reference: https://github.com/DIYgod/RSSHub/discussions/8002
-->

## 该 PR 相关 Issue / Involved issue

Close #

## 完整路由地址 / Example for the proposed route(s)

<!--
请在 `routes` 区域填写以 / 开头的完整路由地址，否则你的 PR 将会被无条件关闭。
如果路由包含在文档中列出可以完全穷举的参数（例如分类），请依次全部列出。

Please include route starts with /, with all required and optional parameters. Fail to comply will result in your pull request being closed automatically.
```route
/some/route
/some/other/route
/dont/use/this/or/modify/it
/use/the/fenced/code/block/below
```
如果你的 PR 与路由无关, 请在 `routes` 区域 填写 `NOROUTE`，而不是直接删除 `routes` 区域。否则你的 PR 将会被无条件关闭。
If your changes are not related to route, please fill in `routes` with `NOROUTE`. Fail to comply will result in your PR being closed.
-->

```routes
/caai/40
/ccf/tfbd/xwdt/tzgg
```

## 新 RSS 检查列表 / New RSS Script Checklist
  
- [x] 新的路由 New Route
  - [x] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [x] 文档说明 Documentation
  - [x] 中文文档 CN
  - [ ] 英文文档 EN
- [x] 全文获取 fulltext
  - [ ] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [x] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [x] 可以解析 Parsed
  - [x] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

## 说明 / Note

tudou027 committed:
增加大数据专家委员会和中国人工智能学会新闻

TonyRL commented on lib/v2/caai/utils.js:
```diff
@@ -0,0 +1,94 @@
+
+const base = 'http://www.caai.cn';
+
+const urlBase = (caty) => url.resolve(base, `/index.php?s=/home/article/index/id/${caty}.html`); // en, cn, (none, for JP)
```
`url.resolve()` [is legacy](https://nodejs.org/api/url.html#urlparseurlstring-parsequerystring-slashesdenotehost). Please use WHATWG URL API instead.

TonyRL commented on lib/v2/caai/utils.js:
```diff
@@ -0,0 +1,94 @@
+    return info;
+};
+
+const configs = {
+    // 关于CAAI
+    40: { title: '条例与法规' },
+    41: { title: '主要领导' },
+    43: { title: '分支机构' },
+    90: { title: '学会党建' },
+    77: { title: '文档下载' },
+
+    // 学会动态
+    46: { title: '学会新闻' },
+    47: { title: '通知公告' },
+    49: { title: '活动预告' },
+    71: { title: '对外合作' },
+    63: { title: '培训动态' },
+
+    // 党建强会
+    79: { title: '党建强会' },
+    115: { title: '党史学习' },
+
+    // 奖励与推荐
+    65: { title: '吴文俊奖' },
+    86: { title: '青托计划' },
+    67: { title: '院士推荐' },
+    91: { title: '成果鉴定' },
+
+    // CAAI资源
+    52: { title: '学会期刊' },
+    53: { title: '学科皮书系列' },
+    54: { title: '年鉴及发展报告' },
+
+    // 会员专区
+    69: { title: '学会会员' },
+    135: { title: '高级会员' },
+    83: { title: '会员荣誉' },
+    74: { title: '单位会员' },
+};
```
You can use `$('.article-list h1').text()` to obtain the title instead of hardcoding them

TonyRL commented on lib/v2/tfbd/index.js:
```diff
@@ -0,0 +1,30 @@
+    const base = utils.urlBase(ctx.params.caty, ctx.params.id);
+    const res = await got.get(base, {
+        https: {
+            rejectUnauthorized: false,
```
This option is only valid when you perform HTTPS requests. However, according to

https://github.com/DIYgod/RSSHub/blob/c0df7898c1b448bb00ce31f2d447476b8264dc70/lib/v2/tfbd/utils.js#L9

you're performing HTTP requests

TonyRL commented on lib/v2/tfbd/index.js:
```diff
@@ -0,0 +1,30 @@
+        item: info.map((e, i) => ({
+            title: e.title,
+            description: utils.renderDesc(details[i].description),
+            pubDate: details[i].pubDate || dateParser(new Date().toISOString()), // No Time for now
```
[Do **not**](https://docs.rsshub.app/joinus/pub-date.html) set `pubDate` if there isn't any.

TonyRL commented on lib/v2/tfbd/index.js:
```diff
@@ -0,0 +1,30 @@
+const utils = require('./utils');
+const got = require('@/utils/got');
+const dateParser = require('@/utils/dateParser');
```
```suggestion
const { parseDate } = require('@/utils/parse-date');
```
Please use `parseDate()` instead

TonyRL commented on lib/v2/tfbd/radar.js:
```diff
@@ -0,0 +1,13 @@
+module.exports = {
+    'tfbd.ccf.org.cn': {
```
```suggestion
    'ccf.org.cn': {
```
This should include the root domain only

TonyRL commented on lib/v2/tfbd/radar.js:
```diff
@@ -0,0 +1,13 @@
+module.exports = {
+    'tfbd.ccf.org.cn': {
+        _name: 'tfbd',
+        '.': [
```
```suggestion
        tfbd: [
```

TonyRL commented on lib/v2/tfbd/utils.js:
```diff
@@ -0,0 +1,87 @@
+    return info;
+};
+
+const configs = {
+    xwdt: {
+        title: '新闻动态', child: {
+            tzgg: { title: '通知公告' },
+            wydt: { title: '委员动态' },
+        }
+    },
+    pphd: {
+        title: '品牌活动', child: {
+            xsjl: { title: '学术交流' },
+            jsjl: { title: '技术交流' },
+            cjds: { title: '创建大赛' },
+            zlyj: { title: '战略研究' },
+            jy: { title: '教育' },
+            top10dsjyy: { title: 'Top10大数据应用' },
+            zjxl: { title: '走进系列' },
+        }
+    },
+    ljhd: {
+        title: '历届活动', child: {
+            xsjl: { title: '学术交流' },
+            jsjl: { title: '技术交流' },
+            cxds: { title: '创新大赛' },
+        }
+    },
+};
```
You can obtain the title from the title tag

TonyRL commented on lib/v2/tfbd/utils.js:
```diff
@@ -0,0 +1,87 @@
+
+const base = 'http://tfbd.ccf.org.cn';
+
+const urlBase = (caty, id) => url.resolve(base, `/tfbd/${caty}/${id}/`);
```
`url.resolve()` [is legacy](https://nodejs.org/api/url.html#urlparseurlstring-parsequerystring-slashesdenotehost). Please use WHATWG URL API instead.

TonyRL reviewed (changes requested):
- Please use the domain name `ccf` instead of `tfbd`
- Missing documentation

tudou027 committed:
Merge branch 'DIYgod:master' into master

tudou027 committed:
修复问题

TonyRL committed:
fix: add docs for `caai`

fix: maintainer does not with match with router

TonyRL committed:
fix: add docs for `ccf/tfbd`

fix: use raw output for `art-template`

fix: standardise use of `got`
not using `got()` in one place and `got.get()` in another place

TonyRL committed:
refactor: migrate to v2
